x-pack/plugins/ml/server/models/job_audit_messages/job_audit_messages.js

```diff
@@ -266,12 +278,68 @@ export function jobAuditMessagesProvider({ asInternalUser }, mlClient) {
     return jobMessages;
   }
 
+  const clearedTime = new Date().getTime();
+
+  // Sets 'cleared' to true for messages in the last 24hrs and index new message for clear action
+  async function clearJobAuditMessages(jobId) {
+    const newClearedMessage = {
+      job_id: jobId,
+      job_type: 'anomaly_detection',
+      level: MESSAGE_LEVEL.INFO,
+      message: 'Cleared set to true for messages in the last 24hrs.',
+      timestamp: clearedTime,
+    };
+
+    const query = {
+      bool: {
+        filter: [
+          {
+            range: {
+              timestamp: {
+                gte: 'now-24h',
+              },
+            },
+          },
+          {
+            term: {
+              job_id: jobId,
+            },
+          },
+        ],
+      },
+    };
+
+    await Promise.all([
+      asCurrentUser.updateByQuery({
```

Comment:
i think these requests should be performed by the internal user, with a suitable permission check on the route

Reply:
Will be updated in a follow up bug fix once https://github.com/elastic/elasticsearch/pull/74711 is in so we can use internal user 👍